docs/content/InputField.mdx

```diff
@@ -0,0 +1,192 @@
+---
+title: InputField
+status: Alpha
+source: https://github.com/primer/react/blob/main/src/InputField.tsx
+storybook: '/react/storybook?path=/story/forms-inputfield--text-input-field'
+---
+
+import {InputField, TextInputWithTokens} from '@primer/react'
+import {ComponentChecklist} from '../src/component-checklist'
+
+The `InputField` component is used to render a labelled text input and, optionally, associated validation text and hint text.
+
+## Anatomy
+
+### InputField
+
+The container that handles the layout and passes the relevant IDs and ARIA attributes it's children.
+
+### InputField.Label
+
+A `InputField.Label` must be passed, but it may be visually hidden.
+
+### InputField.Caption
+
+If this field needs additional context, a `InputField.Caption` may be used to render hint text.
+
+### InputField.Validation
+
+If the field was flagged during validation, `InputField.Validation` may be used to render contextual validation information to help the user complete their task.
+
+## Examples
+
+### Basic
+
+```jsx live
+<InputField>
+  <InputField.Label>Name</InputField.Label>
+  <TextInput />
+</InputField>
+```
+
+### Required
+
+```jsx live
+<InputField required>
+  <InputField.Label>Name</InputField.Label>
+  <TextInput />
+</InputField>
+```
+
+### Disabled
+
+```jsx live
+<InputField disabled>
+  <InputField.Label>Name</InputField.Label>
+  <TextInput />
+</InputField>
+```
+
+### Using a TextInputWithTokens input
+
+```javascript live noinline
+const TextInputWithTokensExample = () => {
+  const [tokens, setTokens] = React.useState([
+    {text: 'zero', id: 0},
+    {text: 'one', id: 1},
+    {text: 'two', id: 2}
+  ])
+  const onTokenRemove = tokenId => {
+    setTokens(tokens.filter(token => token.id !== tokenId))
+  }
+
+  return (
+    <InputField>
+      <InputField.Label>Name</InputField.Label>
+      <TextInputWithTokens onTokenRemove={onTokenRemove} tokens={tokens} />
+    </InputField>
+  )
+}
+
+render(TextInputWithTokensExample)
+```
+
+### With a visually hidden label
+
+```jsx live
+<InputField>
+  <InputField.Label visuallyHidden>Name</InputField.Label>
+  <TextInput />
+</InputField>
+```
+
+### With a caption
+
+```jsx live
+<InputField>
+  <InputField.Label>Name</InputField.Label>
+  <TextInput />
+  <InputField.Caption>Hint: your first name</InputField.Caption>
+</InputField>
+```
+
+### With validation
+
+```javascript live noinline
```

Comment:
this example displays strangely for me, does it look like this for you @mperrotti? 

![Screenshot 2021-12-15 at 15 16 44](https://user-images.githubusercontent.com/13340707/146213291-f32da90e-c8b2-46b9-90f6-96cd89ce5fa6.png)

Reply:
It does. Easy fix though :)